4337/contracts/Safe4337Module.sol

```diff
@@ -166,6 +172,7 @@ contract Safe4337Module is IAccount, HandlerContext, CompatibilityFallbackHandle
      * @return validationData An integer indicating the result of the validation.
      */
     function _validateSignatures(address entryPoint, UserOperation calldata userOp) internal view returns (uint256 validationData) {
+        (uint96 signatureTimestamps, bytes memory signatures) = abi.decode(userOp.signature, (uint96, bytes));
```

Comment:
Sorry to be annoying here 🙈.

I think it might be worth having a discussion on what we want to use for the signature format for ERC-4337 signatures.

Personally, I do agree with the choice, but I would argue it is not obvious:
- Packed encoding? It might save some gas (might be worth experimenting at least)
- More complicated encoding (similar to the 4-byte selector scheme that we had for the `SignatureManager` in the protocol, where if the signature starts with `bytes4(keccak(“SignatureWithValidity(uint96,bytes)”))` then we parse it this way, otherwise we use `signatureTimestamps = 0` and the signature bytes be the signature bytes).

One benefit of manual decoding of the signature, is that we can pass `memory signatures` to the `getOperationHash` function, but keep it `bytes calldata` for the `checkSignatures(operationHash, "", signatures)` call to avoid a loop for copying memory (not 100% sure this will have an impact, but I assume it is similar to the gas optimization you made in the execution function).

Reply:
As discussed in the standup, I've implemented the packed encoding. We decided not to implement the 4-byte selector as it would be very complicated.